File: internal/dto/issue.go
package dto

type IssueOption struct {
	Value string `json:"value"`
	Label string `json:"label"`
}

type IssueMeta struct {
	ProblemTypes  []IssueOption `json:"problemTypes"`
	HandleMethods []IssueOption `json:"handleMethods"`
	Statuses      []IssueOption `json:"statuses"`
}

type IssueLookupResult struct {
	OrderNo             string `json:"orderNo"`
	PlatformOrderID     string `json:"platformOrderId,omitempty"`
	PlatformAftersaleID string `json:"platformAftersaleId,omitempty"`
	ShopID              uint64 `json:"shopId,omitempty"`
	ShopName            string `json:"shopName"`
	ProductTitle        string `json:"productTitle"`
	ProductImage        string `json:"productImage,omitempty"`
	SkuSpecs            string `json:"skuSpecs,omitempty"`
	BuyerName           string `json:"buyerName,omitempty"`
	BuyerPhone          string `json:"buyerPhone,omitempty"`
	BuyerAddress        string `json:"buyerAddress,omitempty"`
	HasAftersale        bool   `json:"hasAftersale"`
	AftersaleStatus     string `json:"aftersaleStatus,omitempty"`
	AftersaleType       string `json:"aftersaleType,omitempty"`
	Logistics           string `json:"logistics,omitempty"`
	ReturnLogisticsNo   string `json:"returnLogisticsNo,omitempty"`
	ShipLogisticsNo     string `json:"shipLogisticsNo,omitempty"`
	Source              string `json:"source"` // ticket / order / both
}

type IssueItem struct {
	ID                  uint64  `json:"id"`
	ShopID              uint64  `json:"shopId,omitempty"`
	ShopName            string  `json:"shopName"`
	OrderNo             string  `json:"orderNo,omitempty"`
	PlatformOrderID     string  `json:"platformOrderId,omitempty"`
	PlatformAftersaleID string  `json:"platformAftersaleId,omitempty"`
	ProductTitle        string  `json:"productTitle"`
	ProductImage        string  `json:"productImage,omitempty"`
	SkuSpecs            string  `json:"skuSpecs,omitempty"`
	BuyerName           string  `json:"buyerName,omitempty"`
	BuyerPhone          string  `json:"buyerPhone,omitempty"`
	BuyerAddress        string  `json:"buyerAddress,omitempty"`
	HasAftersale        bool    `json:"hasAftersale"`
	AftersaleStatus     string  `json:"aftersaleStatus,omitempty"`
	AftersaleType       string  `json:"aftersaleType,omitempty"`
	Logistics           string  `json:"logistics,omitempty"`
	ReturnLogisticsNo   string  `json:"returnLogisticsNo,omitempty"`
	ShipLogisticsNo     string  `json:"shipLogisticsNo,omitempty"`
	ProblemType         string  `json:"problemType"`
	ProblemTypeCustom   string  `json:"problemTypeCustom,omitempty"`
	ProblemTypeLabel    string  `json:"problemTypeLabel"`
	HandleMethod        string  `json:"handleMethod,omitempty"`
	HandleMethodCustom  string  `json:"handleMethodCustom,omitempty"`
	HandleMethodLabel   string  `json:"handleMethodLabel,omitempty"`
	ProblemNote         string  `json:"problemNote,omitempty"`
	HandleNote          string  `json:"handleNote,omitempty"`
	Status              string  `json:"status"`
	StatusLabel         string  `json:"statusLabel"`
	OperatorName        string  `json:"operatorName,omitempty"`
	CompletedAt         *string `json:"completedAt,omitempty"`
	CreatedAt           string  `json:"createdAt"`
	UpdatedAt           string  `json:"updatedAt"`
}

type IssueListQuery struct {
	ShopID      uint64
	Status      string
	ProblemType string
	Keyword     string
	Page        int
	PageSize    int
}

type IssueUpsertRequest struct {
	ShopID              uint64 `json:"shopId"`
	ShopName            string `json:"shopName"`
	OrderNo             string `json:"orderNo"`
	PlatformOrderID     string `json:"platformOrderId"`
	PlatformAftersaleID string `json:"platformAftersaleId"`
	ProductTitle        string `json:"productTitle"`
	ProductImage        string `json:"productImage"`
	SkuSpecs            string `json:"skuSpecs"`
	BuyerName           string `json:"buyerName"`
	BuyerPhone          string `json:"buyerPhone"`
	BuyerAddress        string `json:"buyerAddress"`
	HasAftersale        bool   `json:"hasAftersale"`
	AftersaleStatus     string `json:"aftersaleStatus"`
	AftersaleType       string `json:"aftersaleType"`
	Logistics           string `json:"logistics"`
	ReturnLogisticsNo   string `json:"returnLogisticsNo"`
	ShipLogisticsNo     string `json:"shipLogisticsNo"`
	ProblemType         string `json:"problemType"`
	ProblemTypeCustom   string `json:"problemTypeCustom"`
	HandleMethod        string `json:"handleMethod"`
	HandleMethodCustom  string `json:"handleMethodCustom"`
	ProblemNote         string `json:"problemNote"`
	HandleNote          string `json:"handleNote"`
	Status              string `json:"status"`
}

type IssueStatusRequest struct {
	Status string `json:"status"`
}

File: web/src/api/issue.ts
import client, { unwrap, type PageData } from './client'

export interface IssueOption {
  value: string
  label: string
}

export interface IssueMeta {
  problemTypes: IssueOption[]
  handleMethods: IssueOption[]
  statuses: IssueOption[]
}

export interface IssueLookup {
  orderNo?: string
  platformOrderId?: string
  platformAftersaleId?: string
  shopId?: number
  shopName: string
  productTitle: string
  productImage?: string
  skuSpecs?: string
  buyerName?: string
  buyerPhone?: string
  buyerAddress?: string
  hasAftersale: boolean
  aftersaleStatus?: string
  aftersaleType?: string
  logistics?: string
  returnLogisticsNo?: string
  shipLogisticsNo?: string
  source: string
}

export interface AftersaleIssue {
  id: number
  shopId?: number
  shopName: string
  orderNo?: string
  platformOrderId?: string
  platformAftersaleId?: string
  productTitle: string
  productImage?: string
  skuSpecs?: string
  buyerName?: string
  buyerPhone?: string
  buyerAddress?: string
  hasAftersale: boolean
  aftersaleStatus?: string
  aftersaleType?: string
  logistics?: string
  returnLogisticsNo?: string
  shipLogisticsNo?: string
  problemType: string
  problemTypeCustom?: string
  problemTypeLabel: string
  handleMethod?: string
  handleMethodCustom?: string
  handleMethodLabel?: string
  problemNote?: string
  handleNote?: string
  status: string
  statusLabel: string
  operatorName?: string
  completedAt?: string | null
  createdAt: string
  updatedAt: string
}

export type IssueUpsertBody = {
  shopId?: number
  shopName: string
  orderNo?: string
  platformOrderId?: string
  platformAftersaleId?: string
  productTitle: string
  productImage?: string
  skuSpecs?: string
  buyerName?: string
  buyerPhone?: string
  buyerAddress?: string
  hasAftersale?: boolean
  aftersaleStatus?: string
  aftersaleType?: string
  logistics?: string
  returnLogisticsNo?: string
  shipLogisticsNo?: string
  problemType: string
  problemTypeCustom?: string
  handleMethod?: string
  handleMethodCustom?: string
  problemNote?: string
  handleNote?: string
  status?: string
}

export async function fetchIssueMeta() {
  return unwrap<IssueMeta>(await client.get('/issue-records/meta'))
}

export async function lookupIssueSource(q: string) {
  return unwrap<{ list: IssueLookup[] }>(await client.get('/issue-records/lookup', { params: { q } }))
}

export async function fetchIssues(params?: {
  shopId?: number
  status?: string
  problemType?: string
  keyword?: string
  page?: number
  pageSize?: number
}) {
  return unwrap<PageData<AftersaleIssue>>(await client.get('/issue-records', { params }))
}

export async function createIssue(body: IssueUpsertBody) {
  return unwrap<AftersaleIssue>(await client.post('/issue-records', body))
}

export async function updateIssue(id: number, body: IssueUpsertBody) {
  return unwrap<AftersaleIssue>(await client.put(`/issue-records/${id}`, body))
}

export async function updateIssueStatus(id: number, status: string) {
  return unwrap<AftersaleIssue>(await client.patch(`/issue-records/${id}/status`, { status }))
}

export async function deleteIssue(id: number) {
  return unwrap<{ ok: boolean }>(await client.delete(`/issue-records/${id}`))
}

/** 处理复制：地址 / 规格 / 问题 / 处理 组合文本 */
export function buildIssueHandleText(row: AftersaleIssue) {
  const lines: string[] = []
  const addr = [row.buyerName, row.buyerPhone, row.buyerAddress].filter(Boolean).join(' ').trim()
  if (addr) lines.push(`地址：${addr}`)
  if (row.skuSpecs?.trim()) lines.push(`规格：${row.skuSpecs.trim()}`)
  else if (row.productTitle?.trim()) lines.push(`商品：${row.productTitle.trim()}`)
  const problem = [row.problemTypeLabel, row.problemNote].filter(Boolean).join(' · ')
  if (problem) lines.push(`问题：${problem}`)
  const handle = [row.handleMethodLabel, row.handleNote].filter(Boolean).join(' · ')
  if (handle) lines.push(`处理：${handle}`)
  if (row.orderNo) lines.push(`订单：${row.orderNo}`)
  if (row.platformAftersaleId) lines.push(`售后：${row.platformAftersaleId}`)
  if (row.shopName) lines.push(`店铺：${row.shopName}`)
  return lines.join('\n')
}

export function buildMergedIssueHandleText(rows: AftersaleIssue[]) {
  return rows.map((r, i) => `【${i + 1}】\n${buildIssueHandleText(r)}`).join('\n\n')
}

export async function copyText(text: string) {
  const value = String(text || '').trim()
  if (!value) return false
  try {
    if (navigator.clipboard?.writeText) {
      await navigator.clipboard.writeText(value)
      return true
    }
  } catch {
    /* fallback */
  }
  const ta = document.createElement('textarea')
  ta.value = value
  ta.style.position = 'fixed'
  ta.style.left = '-9999px'
  document.body.appendChild(ta)
  ta.select()
  const ok = document.execCommand('copy')
  document.body.removeChild(ta)
  return ok
}
